src/setup_payload/java/src/chip/setuppayload/SetupPayloadParser.java

```diff
@@ -4,20 +4,31 @@
 public class SetupPayloadParser {
 
   /** Returns {@link SetupPayload} parsed from the QR code string. */
-  public SetupPayload parseQrCode(String qrCodeString) {
+  public SetupPayload parseQrCode(String qrCodeString) throws UnrecognizedQrCodeException {
     return fetchPayloadFromQrCode(qrCodeString);
   }
 
   /** Returns {@link SetupPayload} parsed from the manual entry code string. */
-  public SetupPayload parseManualEntryCode(String entryCodeString) {
+  public SetupPayload parseManualEntryCode(String entryCodeString)
+      throws UnrecognizedQrCodeException {
```

Comment:
`UnrecognizedQrCodeException` doesn't seem like the right exception type to throw here. What does the entry code string look like? Perhaps something like `InvalidEntryCodeFormatException`?